llvm/test/Transforms/Attributor/nodelete.ll

```diff
@@ -10,15 +10,14 @@ define hidden i64 @f1() align 2 {
 ; TUNIT-LABEL: define {{[^@]+}}@f1
 ; TUNIT-SAME: () #[[ATTR0:[0-9]+]] align 2 {
 ; TUNIT-NEXT:  entry:
-; TUNIT-NEXT:    [[REF_TMP1:%.*]] = alloca i8, i32 0, align 8
+; TUNIT-NEXT:    [[REF_TMP1:%.*]] = alloca [0 x i8], align 1
```

Comment:
I wonder if we want to take a step further to eliminate this instruction?

Reply:
I would think AAIsDead should be able to remove it, However it seems like it is not implemented atm. 
I think this should be taken care of when the implementation of AAIsDead is extended to keep track of each instruction.